duckduckgo/iOS
# Add large scale https updates with bloom filter

Task/Issue URL: https://app.asana.com/0/72649045549333/721626625236431
Tech Design URL: https://app.asana.com/0/72649045549333/727813825116580

**Description**:
Add large scale https support:
- Remove legacy https solution
- Add updated https solution
- Add new api requests 
- Update core data model and persitence layer to support new data

**Steps to test this PR**:

TEST ONE:
1. Visit http://example.com
1. Check the logs and ensure they report that this site is "upgradable"
1. Note that the scheme is updated to https

TEST TWO:
1. Visit http://aaaaaaaaaaaaa.co.uk
1. Check the logs and ensure they report that this site is not "upgradable"
1. Note that the scheme remains as "http"
1. The site itself doesn't exist (thus definitely not in our upgrade list), so won't load

TEST THREE:
1. Visit any site from https://staticcdn.duckduckgo.com/https/https-mobile-whitelist.json
1. Note the logs stats that it is whitelisted and cannot be autoupgraded
1. Note that the scheme remains as "http" (beware some sites will upgrade themselves!)

TEST FOUR:

1. Install the app from develop and run it
2. Then install this version over the top
3. Ensure that nothing in the db goes boom!

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
